src/link.zig

```diff
@@ -282,6 +283,13 @@ pub const HashStyle = enum { sysv, gnu, both };
 
 pub const CompressDebugSections = enum { none, zlib };
 
+pub const LibCProvider = enum {
+    none,
+    installation,
+    sysroot,
+    vendored,
+};
```

Comment:
Can you add comments to each variant explaining its meaning?

Reply:
I am very glad you brought this up. It caused me to focus the changes and provide clarity.
Some of the confusion was a result of my overloading the LibCProvider struct for future non-macos uses. This is not the right time for that.

- renamed `LibCProvider` → `DarwinSdkLayout`
- documented the struct
- dropped `installation` (which used to mean LibCInstallation mechanis) because code already transforms it to the sysroot method